### ISSUE TITLE
I cannot add a tool to toolbar in Overlay

### ISSUE THREAD

**Issue by euacar** [ioam/holoviews#2818]
I cannot add other tools to the output html. 
tools options is not working assuming that I am setting correctly 

I also tried with specific; hover = HoverTool(tooltips=[("%",'@Percent'),("WW","@WW"),("A","@AREA")])

------

```
curve = hv.Curve(df,key_dimensions,value_dimensions).groupby(value_dimensions).overlay()
points = hv.Points(df,key_dimensions,value_dimensions).groupby(value_dimensions).overlay()

curve_points = curve * points
curve_points = curve_points.opts(
    plot=dict(
        width=1500,
        height=500,
        tools=['hover','yzoom_in'],
        toolbar='above',
        title_format="Usage by A"))


finalplot = curve_points


plot = renderer.get_plot(finalplot).state

save(plot,"Out.html",title="Out",resources= bk.resources.INLINE,logo=None)
```

**Comment by philippjfr** [comment]
I think there might some some bug here but first I need to figure out what you're trying to do. It looks like you are applying a groupby along the value dimensions which is not supported as far as I know. Secondly you'll likely have to apply the hover tool to the points and curve individually, since it needs to associate the tooltips with those particular plot elements.

**Comment by ahuang11** [comment]
I had issues adding/modifying the existing tools too (places I dumped `.options(tools=TOOLS)`, but I can't seem to reproduce it with a simple example...

```
import holoviews as hv
import pandas as pd
import numpy as np
import parambokeh
import param

hv.extension('bokeh')

def cpd_int(p, r, t=840, n=12, c=0, cf=1, cr=0, crf=1):
    """
    Calculate compound interest
    :param p: (float) - principal or initial deposit
    :param r: (float) - annual interest rate in percent
    :param t: (int) - time in months
    :param n: (int) - number of times interest is compounded each year
    :param c: (float) - recurring contribution at the beginning of every cf
    :param cf: (int) - contribution frequency in months
    :param cr: (float) - additional contributions added every crf
    :param crf: (int) - contribution ramp frequency in months
    :return p: (float) - final amount
    """
    nom = int(1 / (n / 12))  # number of months 
    r /= 100
    p_arr = np.zeros(t)

    tot_c = 0
    for i in range(t):
        if i % crf == 0:
            c += cr
        if i % cf == 0:
            p += c
            tot_c += c
        if i % nom == 0:
            p *= (1 + r / n)
            p_arr[i] = p

    tot_r = p - tot_c - p_arr[0]
    return p_arr, tot_c, tot_r

TOOLS = ['reset']
CURVE_OPTS = dict(
    width=925,
    height=500,
    color='black',
    show_grid=True,
    show_legend=False,
    default_tools=[],
    line_width=0.75,
    tools=TOOLS,
    logy=True,
)

class InterChange(hv.streams.Stream):
    Initial_savings_in_dollars = param.Number(default=1000, bounds=(0, 100000), precedence=0)
    Annual_interest_rate_in_percent = param.Number(default=1.65, bounds=(0, 10), precedence=2)
    Contributions_in_dollars = param.Number(default=0, bounds=(0., 20000.), precedence=3)
    Contribution_frequency_in_months = param.Number(default=12, bounds=(1., 240), precedence=4)
    Increasing_contributions_in_dollars = param.Number(default=0, bounds=(0., 5000.), precedence=5)
    Increasing_contribution_frequency_in_months = param.Number(default=24, bounds=(1., 240.), precedence=6)

    output = parambokeh.view.Plot()
    
    def view(self, audit_dt=None, *args, **kwargs):
        p_arr, tot_c, tot_r = cpd_int(
            self.Initial_savings_in_dollars,
            self.Annual_interest_rate_in_percent,
            c=self.Contributions_in_dollars,
            cf=self.Contribution_frequency_in_months,
            cr=self.Increasing_contributions_in_dollars,
            crf=self.Increasing_contribution_frequency_in_months
        )
        tot_p = p_arr[-1]
        ini_dt = pd.datetime.today()

        dts = pd.date_range(ini_dt,
                            periods=len(p_arr),
                            freq='1MS')
        df = pd.DataFrame({
            'Time': dts,
            'Savings [$]': p_arr
        })

        curve = hv.Curve(df, 'Time', 'Savings [$]')
        curve = curve.options(**CURVE_OPTS)
        curve = curve.redim.range(**{'Savings [$]': (
            self.Initial_savings_in_dollars, self.Initial_savings_in_dollars ** 2.5)})
        
        if audit_dt is None:
            audit_dt = ini_dt + pd.Timedelta(days=365)
        else:
            audit_dt = np.clip(audit_dt, dts[0], dts[-1])
        vline = hv.VLine(audit_dt).options(color='red', line_width=0.5)

        months_diff = int((audit_dt - ini_dt).days / 31)
        tgt_p, tgt_c, tgt_r = cpd_int(
            self.Initial_savings_in_dollars,
            self.Annual_interest_rate_in_percent,
            t=months_diff,
            c=self.Contributions_in_dollars,
            cf=self.Contribution_frequency_in_months,
            cr=self.Increasing_contributions_in_dollars,
            crf=self.Increasing_contribution_frequency_in_months
        )
        tgt_p = tgt_p[-1]

        years_diff = int(months_diff / 12)
        text = '{0} years: Total ${1:,.0f}, Interest ${2:,.0f}, Contributions ${3:,.0f}, Initial ${4:,.0f}'.format(
            years_diff, tgt_p, tgt_r, tgt_c, self.Initial_savings_in_dollars)

        return (curve.relabel(text) * vline).cols(1).options(tools=TOOLS)

    def event(self, **kwargs):
        if not self.output:
            streams = [self, hv.streams.PointerX().rename(x='audit_dt')]
            self.output = hv.DynamicMap(self.view, streams=streams).options(tools=TOOLS)
        else:
            super(InterChange, self).event(**kwargs)

interchange = InterChange(name='')
parambokeh.Widgets(interchange, callback=interchange.event, push=False,
                   on_init=True, view_position='below', width=925)
```

**Comment by euacar** [comment]
If I changed to below, it works as expected. I was trying to make it more concise.. Thx 
---------------
curve = hv.Curve(df,key_dimensions,value_dimensions).groupby(value_dimensions)
curve = (curve.opts(plot=plot_options)).overlay()

points = hv.Points(df,key_dimensions,value_dimensions).groupby(value_dimensions)
points = (points.opts(plot=plot_options)).overlay()

curve_points = curve * points

**Comment by euacar** [comment]
now I having a problem with 
grid_style = {'grid_line_color': 'black', 'grid_line_width': 10,'ygrid_line_dash':[10,10]}
.. It does not take it.

If I have only Curve or points without any overlay.. no issue.


grid_style = {'grid_line_color': 'black', 'grid_line_width': 10,'ygrid_line_dash':[10,10]}
plot_options = dict(
        width=1500,
        height=500,
        sizing_mode='scale_width',
        tools=[hover,'yzoom_in'],
        toolbar='above',
        title_format="L8 Usage by FA",
        xticks=WWs,
        yticks=percents,
        gridstyle=grid_style,
        legend_position='bottom_right',
        show_grid= True,
        fontsize={'ticks':'15pt','title':'30pt'})


curve = hv.Curve(df,key_dimensions,value_dimensions).groupby(value_dimensions)
curve = (curve.opts(plot=plot_options)).overlay()

points = hv.Points(df,key_dimensions,value_dimensions).groupby(value_dimensions)
points = (points.opts(plot=plot_options)).overlay()

curve_points = curve * points

**Comment by philippjfr** [comment]
@euacar Could you confirm your version of holoviews? I would also recommend using ``.options`` over ``.opts`` because it's less nested and performs immediate validation.

**Comment by euacar** [comment]
holoviews==1.10.1
bokeh==0.12.15

**Comment by euacar** [comment]
I changed to .options as below and got the error below and no gridstyle still. .opts does not have a problem of positioning the legend.
ValueError: Unexpected option 'legend_position' for Curve types across all extensions. No similar options found.


curve = hv.Curve(df,key_dimensions,value_dimensions).groupby(value_dimensions)
curve = (curve.options(**plot_options)).overlay()

points = hv.Points(df,key_dimensions,value_dimensions).groupby(value_dimensions)
points = (points.options(**plot_options)).overlay()

**Comment by philippjfr** [comment]
A fix for gridstyles was merged as part of 1.10.5 (see https://github.com/ioam/holoviews/pull/2765), so try updating.

**Comment by euacar** [comment]
thx... new version solved the gridstyle problem...
